### PR TITLE
docs: Open graph image preview internal docs

### PR DESCRIPTION
**Pull request by SiTaggart**
Document how the open graph image preview function works

Best reviewed if you view the file 

image

### CODE REVIEW

File: internal-docs/engineering/doc-site/open-graph-image-preview-function.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+yarn start:website
+```
+
+Once both the website and netlify functions have started running, both will be available on `localhost:8888`.
+
+🚨 **It takes some time for Gastby to boot up. Be patient** 🚨
+
+Once running, the function can be reached at `localhost:8888/.netlify/functions/opengraph/` for testing.
+
+Whilst running, changes to the function code will restart the function with the new code changes applied. (Hot reloading for Node, using Nodemon) This is all handled for us using the Netlify CLI.
```
Comment (TheSisb) [inline internal-docs/engineering/doc-site/open-graph-image-preview-function.md:55]: Can we leave a note about where the code lives around here?

> The code that generates the opengraph preview page lives in:
> - packages/paste-website/functions/opengraph.ts (the function code)
> - packages/paste-website/src/pages/opengraph/index.tsx (the code for the UI that is screenshotted)
> - packages/paste-website/stories/Opengraph.stories.tsx (a story to aid in local development for the UI)

File: internal-docs/engineering/doc-site/open-graph-image-preview-function.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Open Graph Image Preview Function
+
+When sharing a link to a component page on the internet, we supply a dynamically created [Open Graph](https://ogp.me/) image to be expanded by the service that the link is being shared on. Services can include Twitter and Slack.
+
+**Example:**
+
+<img width="570" alt="image" src="https://user-images.githubusercontent.com/368249/182693326-5068e8a4-1e86-432e-81ae-b19e07a53905.png">
```
Comment (TheSisb) [inline internal-docs/engineering/doc-site/open-graph-image-preview-function.md:7]: This is very excellent. 👏  How did you generate it?

**Review by TheSisb — APPROVED** [review]

File: internal-docs/engineering/doc-site/open-graph-image-preview-function.md (hunk 2 above)
Comment (shleewhite) [inline internal-docs/engineering/doc-site/open-graph-image-preview-function.md:7]: Suggestion: you can probably have better alt text here (or none at all)

File: internal-docs/engineering/doc-site/open-graph-image-preview-function.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+## The way it works
+
+```mermaid
```
Comment (shleewhite) [inline internal-docs/engineering/doc-site/open-graph-image-preview-function.md:11]: Praise: this is cool!

File: internal-docs/engineering/doc-site/open-graph-image-preview-function.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+[Winston](https://github.com/winstonjs/winston) does our logging, so we can observe the function running in real time in the Netlify UI.
+
+[Rollbar](https://docs.rollbar.com/docs/aws-lambda) wraps our function and helps log errors to the Rollbar service so we are alerted if it starts to break.
+
+## Node runtime 
+
+🚨 **Node v14** 🚨
+
+The Netlify function must run using Node v14 for the Chrome binary to work correctly. We are explicitly opting out of Node 16 runtime in Netlify functions by setting the `AWS_LAMBDA_JS_RUNTIME` environment variable in the Netlify UI.
```
Comment (shleewhite) [inline internal-docs/engineering/doc-site/open-graph-image-preview-function.md:75]: Question: what would it take for the Chrome binary to work with Node 16? Is this work that we will have to do later?

Comment (SiTaggart) [reply internal-docs/engineering/doc-site/open-graph-image-preview-function.md:75]: There is a JIRA ticket for it now https://issues.corp.twilio.com/browse/DSYS-3479. What needs to be done needs to be figured out

**Review by shleewhite — APPROVED** [review]

**Pull request merged**